packages/cdk8s-cli/src/cli/cmds/synth.ts

```diff
@@ -12,37 +13,34 @@ class Command implements yargs.CommandModule {
 
   public readonly builder = (args: yargs.Argv) => args
     .option('app', { default: config.app, required: true, desc: 'Command to use in order to execute cdk8s app', alias: 'a' })
-    .option('output', { default: config.output, required: true, desc: 'Output directory', alias: 'o' });
+    .option('output', { default: config.output, required: false, desc: 'Output directory', alias: 'o' })
+    .option('stdout', { type: 'boolean', required: false, desc: 'Write synthesized manifests to STDOUT instead of the output directory', alias: 'p' });
 
   public async handler(argv: any) {
     const command = argv.app;
     const outdir = argv.output;
+    const stdout = argv.stdout;
+
+    if (outdir !== config.output && stdout) {
+      throw new Error('\'--output\' and \'--stdout\' are mutually exclusive. Please only use one.');
+    }
 
     await fs.remove(outdir);
 
-    await shell(command, [], {
-      shell: true,
-      env: {
-        ...process.env,
-        CDK8S_OUTDIR: outdir,
-      },
-    });
-
-    if (!await fs.pathExists(outdir)) {
-      console.error(`ERROR: synthesis failed, app expected to create "${outdir}"`);
-      process.exit(1);
-    }
+    if (stdout) {
+      await mkdtemp(async tempDir => {
```

Comment:
Would that remote the temporary directory after we are done with it?

Reply:
Yep! This is what `mkdtemp` does:

```
  const workdir = await fs.mkdtemp(path.join(os.tmpdir(), 'cdk8s-'));
  try {
    await closure(workdir);
  } finally {
    await fs.remove(workdir);
  }
```